### ISSUE TITLE
[Dynamo] Graph break in Huggingface model Vilt

### ISSUE THREAD

**Issue by BowenBao** [pytorch/pytorch#98045]
### 🐛 Describe the bug

```python
from transformers import ViltProcessor, ViltForQuestionAnswering
from PIL import Image
import requests
import torch.onnx

url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)
text = "How many cats are there?"

processor = ViltProcessor.from_pretrained("dandelin/vilt-b32-finetuned-vqa")
model = ViltForQuestionAnswering.from_pretrained("dandelin/vilt-b32-finetuned-vqa")

inputs = processor(image, text, return_tensors="pt")

import torch._dynamo

gm, _ = torch._dynamo.export(model, **inputs, aten_graph=True, tracing_mode="real")
gm.print_readable()
```

Logs
```
Traceback (most recent call last):
  File "/home/bowbao/pytorch_gh/torch/_dynamo/utils.py", line 1236, in run_node
    return node.target(*args, **kwargs)
  File "/home/bowbao/pytorch_gh/torch/nn/functional.py", line 3968, in interpolate
    return torch._C._nn.upsample_bilinear2d(input, output_size, align_corners, scale_factors)
TypeError: upsample_bilinear2d() received an invalid combination of arguments - got (FakeTensor, tuple, bool, NoneType), but expected one of:
 * (Tensor input, tuple of ints output_size, bool align_corners, tuple of floats scale_factors)
      didn't match because some of the arguments have invalid types: (FakeTensor, tuple of (FakeTensor, FakeTensor), bool, NoneType)
 * (Tensor input, tuple of ints output_size, bool align_corners, float scales_h, float scales_w, *, Tensor out)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/bowbao/pytorch_gh/torch/_dynamo/utils.py", line 1186, in get_fake_value
    return wrap_fake_exception(
  File "/home/bowbao/pytorch_gh/torch/_dynamo/utils.py", line 827, in wrap_fake_exception
    return fn()
  File "/home/bowbao/pytorch_gh/torch/_dynamo/utils.py", line 1187, in <lambda>
    lambda: run_node(tx.output, node, args, kwargs, nnmodule)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/utils.py", line 1248, in run_node
    raise RuntimeError(
RuntimeError: Failed running call_function <function interpolate at 0x7fe887265dc0>(*(FakeTensor(FakeTensor(..., device='meta', size=(1, 768, 12, 12), grad_fn=<ViewBackward0>), cpu),), **{'size': (FakeTensor(FakeTensor(..., device='meta', size=(), dtype=torch.int64), cpu), FakeTensor(FakeTensor(..., device='meta', size=(), dtype=torch.int64), cpu)), 'mode': 'bilinear', 'align_corners': True}):
upsample_bilinear2d() received an invalid combination of arguments - got (FakeTensor, tuple, bool, NoneType), but expected one of:
 * (Tensor input, tuple of ints output_size, bool align_corners, tuple of floats scale_factors)
      didn't match because some of the arguments have invalid types: (FakeTensor, tuple of (FakeTensor, FakeTensor), bool, NoneType)
 * (Tensor input, tuple of ints output_size, bool align_corners, float scales_h, float scales_w, *, Tensor out)

(scroll up for backtrace)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "repro_march_dynamo_shape.py", line 34, in <module>
    gm, _ = torch._dynamo.export(model, **inputs, aten_graph=True, tracing_mode="real")
  File "/home/bowbao/pytorch_gh/torch/_dynamo/eval_frame.py", line 708, in export
    result_traced = opt_f(*args, **kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/eval_frame.py", line 101, in __call__
    return self.dynamo_ctx(self._orig_mod.__call__)(*args, **kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/eval_frame.py", line 237, in _fn
    return fn(*args, **kwargs)
  File "/home/bowbao/pytorch_gh/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/eval_frame.py", line 374, in catch_errors
    return callback(frame, cache_size, hooks)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/convert_frame.py", line 111, in _fn
    return fn(*args, **kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/convert_frame.py", line 276, in _convert_frame_assert
    return _compile(
  File "/home/bowbao/pytorch_gh/torch/_dynamo/utils.py", line 166, in time_wrapper
    r = func(*args, **kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/convert_frame.py", line 341, in _compile
    out_code = transform_code_object(code, transform)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/bytecode_transformation.py", line 530, in transform_code_object
    transformations(instructions, code_options)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/convert_frame.py", line 328, in transform
    tracer.run()
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 1857, in run
    super().run()
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 604, in run
    and self.step()
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 564, in step
    getattr(self, inst.opname)(inst)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 347, in wrapper
    return inner_fn(self, inst)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 1052, in CALL_FUNCTION_KW
    self.call_function(fn, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 495, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/bowbao/pytorch_gh/torch/_dynamo/variables/nn_module.py", line 269, in call_function
    return tx.inline_user_function_return(
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 531, in inline_user_function_return
    result = InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 1950, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 2014, in inline_call_
    tracer.run()
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 604, in run
    and self.step()
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 564, in step
    getattr(self, inst.opname)(inst)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 347, in wrapper
    return inner_fn(self, inst)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 1040, in CALL_FUNCTION_EX
    self.call_function(fn, argsvars.items, kwargsvars.items)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 495, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/bowbao/pytorch_gh/torch/_dynamo/variables/functions.py", line 297, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/variables/functions.py", line 265, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/variables/functions.py", line 98, in call_function
    return tx.inline_user_function_return(
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 531, in inline_user_function_return
    result = InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 1950, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 2014, in inline_call_
    tracer.run()
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 604, in run
    and self.step()
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 564, in step
    getattr(self, inst.opname)(inst)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 347, in wrapper
    return inner_fn(self, inst)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 1052, in CALL_FUNCTION_KW
    self.call_function(fn, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 495, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/bowbao/pytorch_gh/torch/_dynamo/variables/nn_module.py", line 269, in call_function
    return tx.inline_user_function_return(
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 531, in inline_user_function_return
    result = InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 1950, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 2014, in inline_call_
    tracer.run()
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 604, in run
    and self.step()
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 564, in step
    getattr(self, inst.opname)(inst)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 347, in wrapper
    return inner_fn(self, inst)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 1040, in CALL_FUNCTION_EX
    self.call_function(fn, argsvars.items, kwargsvars.items)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 495, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/bowbao/pytorch_gh/torch/_dynamo/variables/functions.py", line 297, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/variables/functions.py", line 265, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/variables/functions.py", line 98, in call_function
    return tx.inline_user_function_return(
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 531, in inline_user_function_return
    result = InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 1950, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 2014, in inline_call_
    tracer.run()
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 604, in run
    and self.step()
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 564, in step
    getattr(self, inst.opname)(inst)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 347, in wrapper
    return inner_fn(self, inst)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 1052, in CALL_FUNCTION_KW
    self.call_function(fn, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 495, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/bowbao/pytorch_gh/torch/_dynamo/variables/functions.py", line 297, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/variables/functions.py", line 265, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/variables/functions.py", line 98, in call_function
    return tx.inline_user_function_return(
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 531, in inline_user_function_return
    result = InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 1950, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 2014, in inline_call_
    tracer.run()
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 604, in run
    and self.step()
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 564, in step
    getattr(self, inst.opname)(inst)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 347, in wrapper
    return inner_fn(self, inst)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 1003, in CALL_FUNCTION
    self.call_function(fn, args, {})
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 495, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/bowbao/pytorch_gh/torch/_dynamo/variables/functions.py", line 98, in call_function
    return tx.inline_user_function_return(
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 531, in inline_user_function_return
    result = InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 1950, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 2014, in inline_call_
    tracer.run()
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 604, in run
    and self.step()
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 564, in step
    getattr(self, inst.opname)(inst)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 347, in wrapper
    return inner_fn(self, inst)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 1052, in CALL_FUNCTION_KW
    self.call_function(fn, args, kwargs)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/symbolic_convert.py", line 495, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/bowbao/pytorch_gh/torch/_dynamo/variables/torch.py", line 587, in call_function
    tensor_variable = wrap_fx_proxy(
  File "/home/bowbao/pytorch_gh/torch/_dynamo/variables/builder.py", line 940, in wrap_fx_proxy
    return wrap_fx_proxy_cls(
  File "/home/bowbao/pytorch_gh/torch/_dynamo/variables/builder.py", line 975, in wrap_fx_proxy_cls
    example_value = get_fake_value(proxy.node, tx)
  File "/home/bowbao/pytorch_gh/torch/_dynamo/utils.py", line 1215, in get_fake_value
    raise TorchRuntimeError() from e
torch._dynamo.exc.TorchRuntimeError: 

from user code:
   File "/home/bowbao/transformers/src/transformers/models/vilt/modeling_vilt.py", line 127, in visual_embed
    [
  File "/home/bowbao/transformers/src/transformers/models/vilt/modeling_vilt.py", line 129, in <listcomp>
    nn.functional.interpolate(

Set torch._dynamo.config.verbose=True or TORCHDYNAMO_VERBOSE=1 for more information


You can suppress this exception and fall back to eager by setting:
    torch._dynamo.config.suppress_errors = True
```

### Versions

torch at 2f86c9bc0b0281074400548283461e241cac1a35
transformers                  4.25.1

cc @ezyang @gchanan @zou3519 @soumith @msaroufim @wconstab @ngimel @bdhirsh @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

**Comment by yanboliang** [comment]
I found several similar issues in 14k github models, it's probably because we don't handle the meta functions correctly for a few ops.

**Comment by yanboliang** [comment]
#97508 probably can fix this.

**Comment by yanboliang** [comment]
I had another repro:
```
import torch
import torch._dynamo

class MyModule(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.register_buffer('n', torch.tensor(4))

    def forward(self, x):
        return torch.ones(self.n, device=x.device)


m = MyModule()
x = torch.rand([10, 10])
opt_m = torch._dynamo.optimize("eager")(m)

print(m(x))
print(opt_m(x))
```

Error:
```
RuntimeError: Failed running call_function <built-in method ones of type object at 0x7f62e8efb720>(*(FakeTensor(FakeTensor(..., device='meta', size=(), dtype=torch.int64), cpu),), **{'device': device(type='cpu')}):
ones() received an invalid combination of arguments - got (FakeTensor, device=torch.device), but expected one of:
 * (tuple of ints size, *, tuple of names names, torch.dtype dtype, torch.layout layout, torch.device device, bool pin_memory, bool requires_grad)
 * (tuple of ints size, *, Tensor out, torch.dtype dtype, torch.layout layout, torch.device device, bool pin_memory, bool requires_grad)
```

**Comment by BowenBao** [comment]
@yanboliang thanks for quick follow up! I tried https://github.com/pytorch/pytorch/pull/97508 locally, the issue persists though. I guess it probably does unblock the `tuple of FakeTensor` use case. What's left blocking is probably the last argument `NoneType`

```
upsample_bilinear2d() received an invalid combination of arguments - got (FakeTensor, tuple, bool, NoneType), but expected one of:
 * (Tensor input, tuple of ints output_size, bool align_corners, tuple of floats scale_factors)
      didn't match because some of the arguments have invalid types: (FakeTensor, tuple of (FakeTensor, FakeTensor), bool, NoneType)
 * (Tensor input, tuple of ints output_size, bool align_corners, float scales_h, float scales_w, *, Tensor out)
```

**Comment by yanboliang** [comment]
Yes, then I think this function may don't have a correct meta registration. Let me take a closer look.

**Comment by anijain2305** [comment]
The original issue is fixed. The new issue is this, which given the complexity should be handled in  a separate issue.

~~~
torch._dynamo.exc.UserError: Tried to use data-dependent value in the subsequent computation. This can happen when we encounter unbounded dynamic value that is unknown during tracing time.You will need to explicitly give hint to the compiler. Please take a look at constrain_as_value OR constrain_as_size APIs
~~~

I am closing this issue for now. Please feel free to reopen if needed.
